src/PowerTools/Handlers/ViewContextHandler.cs

```diff
@@ -30,33 +31,55 @@ public void ViewContext(MenuCommand menuCommand, dynamic context, Type systemCon
 
             try
             {
-                var filePath = Path.Combine(
-                    Path.GetTempPath(),
-                    contextType.Name
-                        + (menuCommand.CommandID.ID == PkgCmdIDList.cmdidViewEntityDataModel
-                            ? FileExtensions.EntityDataModel
-                            : FileExtensions.Xml));
+                // Use a unique sub-folder (the project GUID) to prevent clashes of same named contexts
+                // from different solutions.
+                var tmpFolder = Path.Combine(Path.GetTempPath(), _package.ProjectGuid.ToString());
+                Directory.CreateDirectory(tmpFolder);
+
+                // Use the context type 'FullName' to distinguish contexts from multiple projects in the same solution
+                var filePath = Path.Combine(tmpFolder, contextType.FullName);
+                filePath = Path.ChangeExtension(filePath,
+                    menuCommand.CommandID.ID == PkgCmdIDList.cmdidViewEntityDataModel
+                        ? FileExtensions.EntityDataModel
+                        : FileExtensions.Xml);
 
                 if (File.Exists(filePath))
                 {
                     File.SetAttributes(filePath, FileAttributes.Normal);
+
+                    // Close the file if already open to eliminate warning dialogs
+                    if (_package.DTE2.ItemOperations.IsFileOpen(filePath))
+                    {
+                        var openDocuments = _package.DTE2.Documents
```

Comment:
Will this not close all open documents?

Reply:
No just the one.  Check the where clause

```
.Where(d => 
        { 
             VisualStudio.Shell.ThreadHelper.ThrowIfNotOnUIThread(); 
             **return d.FullName == filePath;** 
        }) 
        .ToList();
```

It would have read ` .Where(d => d.FullName == filePath)` if it wasn't for that assertion.